### PR TITLE
Adds optional arguments blacklist and whitelist to koa-ratelimit type definitions

### PR DESCRIPTION
**Pull request by Driptap**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [koa-ratelimit/index.js](https://github.com/koajs/ratelimit/blob/master/index.js#L30)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by OutdatedVersion — APPROVED** [review]

**Comment by OutdatedVersion** [comment]
After fixing the whitespace thing this looks good! Thank you!

**Comment by dislick** [comment]
I'm on my phone so I just had a quick glance at it, but doesn't koa-ratelimit also allow async black/whitelist functions?

**Comment by Driptap** [comment]
> I'm on my phone so I just had a quick glance at it, but doesn't koa-ratelimit also allow async black/whitelist functions?

@dislick 

Yep - I missed that, they do.
I've changed the return types to promises

File: types/koa-ratelimit/koa-ratelimit-tests.ts [hunk 1]
```diff
@@ -18,6 +18,13 @@ app.use(rateLimit({
     db: new Map()
 }));
 
+app.use(rateLimit({
+    driver: 'memory',
+    db: new Map(),
+    blacklist: (context) => Promise.resolve(true),
+    whitelist: (context) => Promise.resolve(false)
+}));
```
Comment (dislick) [inline types/koa-ratelimit/koa-ratelimit-tests.ts:26]: I think the tests should include both synchronous and asynchronous implementations. Something like:

```
blacklist: async (ctx) => true,
whitelist: (ctx) => false
```

File: types/koa-ratelimit/index.d.ts [hunk 2]
```diff
@@ -80,6 +80,16 @@ declare namespace KoaRatelimit {
          * A relation of header to the header's display name.
          */
         headers?: HeaderNameOptions;
+
+        /**
+         * If function returns true, middleware exits before limiting
+         */
+        whitelist?: (context: Context) => Promise<boolean>;
```
Comment (dislick) [inline types/koa-ratelimit/index.d.ts:87]: Consider using a union return type to also support synchronous functions: `boolean | Promise<boolean>`

**Review by dislick — CHANGES_REQUESTED** [review]

**Review by dislick — APPROVED** [review]

**Review by OutdatedVersion — APPROVED** [review]

**Comment by Driptap** [comment]
Thanks peeps 🙏🙏

**Comment by Driptap** [comment]
Ready to merge

**Pull request merged**